### PR TITLE
Remove proning feature

### PR DESCRIPTION
**Pull request by Paneedah**
## 📝 Description

This PR aims to remove everything relevant to the player proning feature.

## 🎯 Goals

Remove the ability for players to prone and remove all trace of proning ever existing.
Okay maybe not completely, but for the most part.

- [x] Remove @ deprecated methods for gun factory builders
- [x] Remove the PlayerAnimations.java class
- [x] Clean up some of the surrounding code
- [x] Remove proning keybind
- [x] Remove proning lang
- [x] Remove proning packets
- [x] Remove the 4 different isProning() methods

## 🚦 Testing 

Tested this by running within dev-env and running around, trying to force myself to prone (external cheat plugin I made before) and done all sorts of actions to make sure no random null pointers exist.

Could not produce any issues other than commit [6a6203b](https://github.com/Cubed-Development/Modern-Warfare-Cubed/commit/6a6203b87192f01d79849c5ed2665454cbd0248b).

## ⏮️ Backwards Compatibility 

Fully backwards compatible upon updating. Players will just be unable to lay down.

## 📖 Added to documentation?

- [ ] 📜 README.md
- [ ] 📑 Documentation
- [ ] 📓 Javadoc
- [ ] 🍕 Comments
- [x] 🙅 No documentation needed

## Summary by CodeRabbit

- **New Features**
	- No new features introduced in this update.
- **Bug Fixes**
	- No specific bug fixes mentioned.
- **Documentation**
	- Updated Changelog to reflect changes.
- **Refactor**
	- Simplified body wearable position adjustments by removing proning-related logic.
	- Streamlined event handling by removing unused imports and message handler registrations.
	- Removed unused proning detection methods, enhancing code maintainability.
	- Improved shell casing positioning logic for better readability.
	- Overhauled player rendering logic for efficiency and clarity.
	- Adjusted key input handling logic, specifically removing proning switch functionality in single-player mode.
- **Chores**
	- No chores specified.
- **Revert**
	- Removed the proning feature in single-player mode, reverting part of the game's mechanics.

### CODE REVIEW

File: src/main/java/com/paneedah/weaponlib/WeaponRenderer.java [hunk 1]
```diff
@@ -3001,158 +2974,137 @@ private class FirstPersonWeaponTransitionProvider implements MultipartTransition
 
 		@Override
 		public List<MultipartTransition<Part, RenderContext<RenderableState>>> getTransitions(RenderableState state) {
-			
-		
 			switch(state) {
-			case MODIFYING:
-				return getSimpleTransition(getBuilder().firstPersonPositioningModifying,
+				case MODIFYING:
```
Comment (Paneedah) [inline src/main/java/com/paneedah/weaponlib/WeaponRenderer.java:2978]: image


Definitely not what I did.

GitHub having a stroke lmfao

File: src/main/java/com/paneedah/weaponlib/compatibility/CompatibleExtraEntityFlags.java [hunk 2]
```diff
@@ -13,13 +13,11 @@
 import net.minecraftforge.common.capabilities.ICapabilitySerializable;
 
 public class CompatibleExtraEntityFlags implements ICapabilitySerializable<NBTBase> {
-    
-    public static int PRONING = 0x01;
+
     public static int FLIP = 0x80000000;
```
Comment (Paneedah) [inline src/main/java/com/paneedah/weaponlib/compatibility/CompatibleExtraEntityFlags.java:17]: No idea what `FLIP` is for @Desoroxxx.

Let me know if you feel this is part of the PRONING.

I haven't been able to really understand.

File: src/main/java/com/paneedah/weaponlib/compatibility/CompatibleExtraEntityFlags.java [hunk 3]
```diff
@@ -118,8 +116,4 @@ public NBTBase serializeNBT() {
     public void deserializeNBT(NBTBase nbt) {
         capabilityContainer.getStorage().readNBT(capabilityContainer, instance, null, nbt);
     }
-
-    public static boolean isProning(EntityPlayer player) {
```
Comment (Paneedah) [inline src/main/java/com/paneedah/weaponlib/compatibility/CompatibleExtraEntityFlags.java:122]: This is like the 4th one of these exact same methods that I've removed lmao.
We have an issue with duplicate code.

**Comment by SticksDeveloper** [comment]
Seems good to me

**Review by SticksDeveloper — APPROVED** [review]
Look's good to me Luna has to review the one review...

**Comment by Koud-Wind** [comment]
When a player is proning, mod will write an NBT with the content `player_entity_flags` to the player. Personally, I think it is necessary to delete `CompatibleExtraEntityFlags`, `EntityControlServerMessageHandler` and `EntityControlClientMessageHandler`

File: src/main/java/com/paneedah/weaponlib/compatibility/CompatibleExtraEntityFlags.java (hunk 2 above)
Comment (Koud-Wind) [reply src/main/java/com/paneedah/weaponlib/compatibility/CompatibleExtraEntityFlags.java:17]: Just flags, I've tried changing it to other values, and proning stops working.

**Comment by Paneedah** [comment]
> When a player is proning, mod will write an NBT with the content `player_entity_flags` to the player. Personally, I think it is necessary to delete `CompatibleExtraEntityFlags`, `EntityControlServerMessageHandler` and `EntityControlClientMessageHandler`

Definitely overlooked those. Thanks for the heads up.

**Review by Koud-Wind — DISMISSED** [review]

File: src/main/java/com/paneedah/mwc/asm/MWCClassTransformer.java [hunk 4]
```diff
@@ -592,12 +574,10 @@ public void visitLdcInsn(Object cst) {
...
 
             super.visitFieldInsn(opcode, owner, name, desc);
-            if (opcode == Opcodes.GETFIELD && (owner.equals("bnl") || owner.equals("net/minecraft/util/MovementInput"))
-                    && (name.equals("jump") || name.equals("g"))) {
+            if (opcode == Opcodes.GETFIELD && (owner.equals("bnl") || owner.equals("net/minecraft/util/MovementInput")) && (name.equals("jump") || name.equals("g"))) {
                 Label l6 = new Label();
                 mv.visitJumpInsn(Opcodes.IFEQ, l6);
                 mv.visitVarInsn(Opcodes.ALOAD, 0);
-                mv.visitMethodInsn(Opcodes.INVOKESTATIC, "com/paneedah/mwc/asm/Interceptors", "isProning", "(Lnet/minecraft/entity/player/EntityPlayer;)Z", false);
                 mv.visitJumpInsn(Opcodes.IFNE, l6);
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/asm/MWCClassTransformer.java:581]: Great crash :trollface:, please do:
```diff
-mv.visitVarInsn(Opcodes.ALOAD, 0);
-mv.visitMethodInsn(Opcodes.INVOKESTATIC, "com/paneedah/mwc/asm/Interceptors", "isProning", "(Lnet/minecraft/entity/player/EntityPlayer;)Z", false);
-mv.visitJumpInsn(Opcodes.IFNE, l6);
```

But I am pretty sure this entire `visitFieldInsn` method is only here for proning so it should be removed.

And test in a real env, running through a normal launcher, get shoot by bandits, see if the game does not crash.

File: src/main/java/com/paneedah/weaponlib/PlayerTransitionProvider.java [hunk 5]
```diff
@@ -3,59 +3,22 @@
...
+        //        figure out this spaghetti mess of fucking MultipartTransition lists and MultipartTransitionProviders. Fuck me sideways
+        //        this is absolute cancer and there is about a 0% chance this is efficient. But oh well - whatever. Gets the job done for now.
+        // TODO: CLEAN THIS UP.
+        return Collections.singletonList(
+                new MultipartTransition<Part, RenderContext<RenderableState>>(Part.MAIN, renderContext -> {}, 200, 0)
+                        .withPartPositionFunction(Part.LEFT_HAND, rc -> {})
+                        .withPartPositionFunction(Part.RIGHT_HAND, rc -> {})
+                        .withPartPositionFunction(Part.LEFT_LEG, rc -> {})
+                        .withPartPositionFunction(Part.RIGHT_LEG, rc -> {})
+        );
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/PlayerTransitionProvider.java:22]: What the hell????
Keep:
```java
return normalTransitions;
```
Without using that constant you would be recreating the transition every time `getTransitions` is called!

File: src/main/java/com/paneedah/weaponlib/PlayerRenderer.java [hunk 6]
```diff
@@ -370,10 +321,10 @@ private void renderHead(Positioner<Part, RenderContext<RenderableState>> positio
         FloatBuffer preBuf = MatrixHelper.getModelViewMatrixBuffer();
         positioner.position(Part.HEAD, renderContext);
         FloatBuffer postBuf = MatrixHelper.getModelViewMatrixBuffer();
-        if(!preBuf.equals(postBuf)) {
+//        if(!preBuf.equals(postBuf)) {
 //            modelPlayer.bipedHead.rotateAngleX = modelPlayer.bipedHead.rotateAngleY = 
 //                    modelPlayer.bipedHead.rotateAngleZ = 0f;
-        }
+//        }
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/PlayerRenderer.java:327]: Unrelated change

File: src/main/java/com/paneedah/weaponlib/PlayerRenderer.java [hunk 7]
```diff
@@ -13,16 +12,16 @@
 import org.lwjgl.opengl.GL11;
 
 import java.nio.FloatBuffer;
-import java.util.List;
 
 public class PlayerRenderer {
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/PlayerRenderer.java:16]: Most of the changes to this class seems uncalled for and not related to this pr goals

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/com/paneedah/mwc/proxies/ClientProxy.java [hunk 8]
```diff
@@ -55,7 +55,7 @@ public void preInit(final MWC mod) {
     public void init(final MWC mod) {
         super.init(mod);
 
-        PlayerAnimations.init(mod);
+        MWC.modContext.setPlayerTransitionProvider(new PlayerTransitionProvider());
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/proxies/ClientProxy.java:58]: Huh????

File: src/main/java/com/paneedah/weaponlib/ClientEventHandler.java [hunk 9]
```diff
@@ -231,9 +228,6 @@ private void update() {
       
 		modContext.getPlayerItemInstanceRegistry().update(player);
 		final PlayerWeaponInstance mainHandHeldWeaponInstance = modContext.getMainHeldWeapon();
-
-		if (PlayerUtil.isProning(player)) PlayerUtil.slowPlayerDown(player, SLOW_DOWN_WHILE_PRONING_ATTRIBUTE_MODIFIER);
-		else PlayerUtil.restorePlayerSpeed(player, SLOW_DOWN_WHILE_PRONING_ATTRIBUTE_MODIFIER);
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/ClientEventHandler.java:236]: Have you checked if those two methods are used anywhere else?

File: src/main/java/com/paneedah/weaponlib/CommonEventHandler.java [hunk 10]
```diff
@@ -298,15 +290,8 @@ protected void setSize(EntityPlayer entityPlayer, float width, float height) {
...
-
-        int updatedFlags = CompatibleExtraEntityFlags.getFlags(event.player);
-        if ((updatedFlags & CompatibleExtraEntityFlags.PRONING) != 0) {
-            // If the player is proning, change their hitbox. TO-DO: Is there a more
-            // efficient way to do this?
-            setSize(event.player, 0.6f, 0.6f); //player.width, player.width);
-        }
+        //if (event.phase != TickEvent.Phase.END)
+        //    return;
     }
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/CommonEventHandler.java:295]: If this is empty then just entirely remove it, we don't need useless event calls

File: src/main/java/com/paneedah/weaponlib/WeaponRenderer.java [hunk 11]
```diff
@@ -169,47 +167,36 @@ public static class Builder {
...
 		private Consumer<RenderContext<RenderableState>> firstPersonPositioningShooting;
-	    private Consumer<RenderContext<RenderableState>> firstPersonPositioningProningShooting;
 
 		private Consumer<RenderContext<RenderableState>> firstPersonPositioningZoomingRecoiled;
 		private Consumer<RenderContext<RenderableState>> firstPersonPositioningZoomingShooting;
 		private Consumer<RenderContext<RenderableState>> firstPersonPositioningLoadIterationCompleted;
 
-		private Consumer<RenderContext<RenderableState>> firstPersonLeftHandPositioning;
-		private Consumer<RenderContext<RenderableState>> firstPersonLeftHandPositioningProning;
+		private Consumer<RenderContext<RenderableState>> firstPersonLeftHandPositioning;;
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/WeaponRenderer.java:181]: Deux points virgules :trollface:

File: src/main/java/com/paneedah/weaponlib/WeaponRenderer.java [hunk 12]
```diff
@@ -2644,42 +2487,15 @@ protected StateDescriptor getFirstPersonStateDescriptor(EntityLivingBase player,
...
 			stateManager = new MultipartRenderStateManager<>(currentState, firstPersonTransitionProvider);
 			firstPersonStateManagers.put(player, stateManager);
 		} else {
-			stateManager.setState(currentState, true, currentState == RenderableState.SHOOTING
-			        || currentState == RenderableState.PRONING_SHOOTING
-					|| currentState == RenderableState.ZOOMING_SHOOTING
-					|| currentState == RenderableState.RUNNING
-					|| currentState == RenderableState.ZOOMING
-					|| currentState == RenderableState.DRAWING);
+			stateManager.setState(currentState, true, currentState == RenderableState.SHOOTING || currentState == RenderableState.ZOOMING_SHOOTING || currentState == RenderableState.RUNNING || currentState == RenderableState.ZOOMING || currentState == RenderableState.DRAWING);
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/WeaponRenderer.java:2498]: Please no, NO GOD PLEASE NO

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/com/paneedah/weaponlib/ClientEventHandler.java (hunk 9 above)
Comment (Paneedah) [reply src/main/java/com/paneedah/weaponlib/ClientEventHandler.java:236]: image

Yes. They are used for changing the player speed when scoped.

File: src/main/java/com/paneedah/mwc/proxies/ClientProxy.java (hunk 8 above)
Comment (Paneedah) [reply src/main/java/com/paneedah/mwc/proxies/ClientProxy.java:58]: Removed class PlayerAnimations, TransitionProvider needs to initialize somewhere still.

**Review by Desoroxxx — APPROVED** [review]

**Review by Desoroxxx — APPROVED** [review]

**Pull request merged**